lib/services/podcast-index/search-show.ts

```diff
@@ -0,0 +1,42 @@
+import { PODCAST_INDEX_BASE_URL, generateHeaders } from './utils';
+
+type Show = {
+  author: string;
+  description: string;
+  episodeCount: number;
+  image: string;
+  language: string;
+  podcastGuid: string;
+  title: string;
+};
+
+type PodcastIndexResponse = {
+  count: number;
+  feeds: Show[];
+};
+
+export const searchShow = async (title: string): Promise<Show> => {
+  const response = await fetch(
+    `${PODCAST_INDEX_BASE_URL}/search/bytitle?q=${title}`,
+    {
+      headers: {
+        Authorization: generateHeaders().Authorization,
+        'User-Agent': 'Beecast',
+        'X-Auth-Date': generateHeaders()['X-Auth-Date'],
+        'X-Auth-Key': generateHeaders()['X-Auth-Key'],
+      } as unknown as Record<string, string>,
```

Comment:
we probably won't need this coercion if we use something like `ofetch` as mentioned here: https://github.com/experiment-station/beecast/pull/24#discussion_r1423639848